fetcher/fetcher_test.go

```diff
@@ -195,3 +196,21 @@ func TestInitializeAsserter(t *testing.T) {
 		})
 	}
 }
+
+func TestNewWithHTTPCLient(t *testing.T) {
+	// Callers can pass an http.Client to
+	// the fetcher via WithClient.
+	// Ensure that the fetcher does not
+	// override it.
+	httpClient := &http.Client{}
+	apiClient := client.NewAPIClient(
+		client.NewConfiguration(
+			"https://serveraddress",
+			DefaultUserAgent,
+			httpClient,
+		),
+	)
+	fetcher := New("https://serveraddress", WithClient(apiClient))
+	var assert = assert.New(t)
+	assert.Same(httpClient, fetcher.rosettaClient.GetConfig().HTTPClient)
```

Comment:
lmk if you think we should assert the Transport field of the httpClient remains the same as well.